network/dag/tree/tree.go

```diff
@@ -14,21 +33,29 @@ type Data interface {
 	// Clone creates an exact copy using the current instance as a prototype.
 	Clone() Data
 	// Insert a new transaction reference.
-	Insert(ref hash.SHA256Hash) error
+	Insert(ref hash.SHA256Hash)
+	// Delete a transaction reference.
+	Delete(ref hash.SHA256Hash)
 	// Add other Data to this one. Returns an error if the underlying datastructures are incompatible.
 	Add(other Data) error
 	// Subtract other Data from this one. Returns an error if the underlying datastructures are incompatible.
 	Subtract(other Data) error
+	// IsEmpty returns true if the concrete type is in its default/empty state.
+	IsEmpty() bool
 	encoding.BinaryMarshaler
 	encoding.BinaryUnmarshaler
 }
 
 // Tree is the interface for an in-memory tree that provides fast access to Data over requested Lamport Clock ranges.
-// Tree is not thread-safe.
 type Tree interface {
 	// Insert a transaction reference at the specified clock value.
 	// The result of inserting the same ref multiple times is undefined.
-	Insert(ref hash.SHA256Hash, clock uint32) error
+	Insert(ref hash.SHA256Hash, clock uint32)
+	// InsertGetDirty inserts a transaction reference like Insert and also returns the dirty leaves like GetUpdates.
+	// This is an atomic operation to make sure no ResetUpdate can happen in between from calling logic.
+	InsertGetDirty(ref hash.SHA256Hash, clock uint32) map[uint32][]byte
+	// Delete a transaction reference without checking if ref is in the Tree
+	Delete(ref hash.SHA256Hash, clock uint32)
```

Comment:
>without checking if ref is in the Tree

What happens if it's not present? If nothing happens, you can leave ouyt this comment imo

Reply:
`Delete` puts -ref in the tree if ref was not in the tree. Deleting a ref from an Xor tree that is not in the tree has the same as effect as `Insert`ing it.